corehq/apps/hqwebapp/forms.py

```diff
@@ -291,12 +291,17 @@ def as_table(self):
 
 
 class HQAuthenticationTokenForm(AuthenticationTokenForm):
+    def __init__(self, user, initial_device, request, **kwargs):
+        super().__init__(user, initial_device, **kwargs)
+        self.request = request
 
     def clean(self):
         try:
             cleaned_data = super(HQAuthenticationTokenForm, self).clean()
         except ValidationError:
-            user_login_failed.send(sender=__name__, credentials={'username': self.user.username})
+            user_login_failed.send(sender=__name__, credentials={'username': self.user.username},
```

Comment:
Is there somewhere that this signal is called with `credentials` from another source (other than an on-the-spot custom made dict)? If not, what do you think about simplifying the argument to `username` rather than `credentials`?
```suggestion
            user_login_failed.send(sender=__name__, username=self.user.username,
```
Probably obvious, but this suggestion cannot be applied in isolation. The idea needs to be pushed out to a few other places in this PR as well.

Reply:
Yes, user_login_failed is a [built-in signal](https://docs.djangoproject.com/en/3.1/ref/contrib/auth/#django.contrib.auth.signals.user_login_failed), which is primarily called when django's built-in authentication fails. The call was duplicated for the token and backup token workflows, so `credentials` needs to stay consistent.